src/services/report.tsx

```diff
@@ -0,0 +1,116 @@
+import { format } from 'date-fns';
+
+import { EGO_JWT_KEY } from 'common/constants';
+import downloader from 'common/downloader';
+import { arrangerProjectId, reportsApiRoot } from 'common/injectGlobals';
+import { ReportConfig } from '../store/reportTypes';
+import isEmpty from 'lodash/isEmpty';
+import { Sqon } from '../store/sqon';
+import { familyMemberAndParticipantIds } from './participants';
+import { TRACKING_EVENTS, trackUserInteraction } from './analyticsTracking';
+import uniq from 'lodash/uniq';
+
+const trackDownload = async (label: string) => {
+  await trackUserInteraction({
+    category: TRACKING_EVENTS.categories.reports,
+    action: TRACKING_EVENTS.actions.download.report,
+    label,
+    value: undefined,
+  });
+};
+
+export const RP_PARTICIPANT_FILE_REPO_KEY = 'participantFileRepo';
+export const RP_CLINICAL_DATA_KEY = 'clinicalData';
+export const RP_FAM_CLINICAL_DATA_KEY = 'familyClinicalData';
+export const RP_FAM_CLINICAL_DATA_FILE_REPO_KEY = 'familyClinicalDataFileRepo';
+export const RP_BIOSPECIMEN_DATA_KEY = 'biospecimenData';
+export const RP_BIOSPECIMEN_FILE_REPO_DATA_KEY = 'biospecimenDataFileRepo';
+
+const entry = reportsApiRoot;
+
+const REPORTS_ROUTES = {
+  [RP_PARTICIPANT_FILE_REPO_KEY]: `${entry}/reports/clinical-data-participants-only`,
+  [RP_CLINICAL_DATA_KEY]: `${entry}/reports/clinical-data`,
+  [RP_FAM_CLINICAL_DATA_KEY]: `${entry}/reports/family-clinical-data`,
+  [RP_FAM_CLINICAL_DATA_FILE_REPO_KEY]: `${entry}/reports/family-clinical-data`,
+  [RP_BIOSPECIMEN_DATA_KEY]: `${entry}/reports/biospecimen-data`,
+  [RP_BIOSPECIMEN_FILE_REPO_DATA_KEY]: `${entry}/reports/biospecimen-data`,
+};
+
+export const shouldCheckAvailability = (reportName: string) =>
+  [RP_FAM_CLINICAL_DATA_FILE_REPO_KEY].includes(reportName);
+
+export const checkAvailability = async (reportName: string, sqon: Sqon): Promise<boolean> => {
+  if (reportName === RP_FAM_CLINICAL_DATA_FILE_REPO_KEY) {
+    const { getFamMembersWithoutPtIds } = await familyMemberAndParticipantIds(sqon);
+    return getFamMembersWithoutPtIds().length > 0;
+  }
+  return true;
+};
+
+export const buildSqonFromFileRepoForReport = async (reportName: string, originalSqon?: Sqon) => {
+  if (reportName === RP_PARTICIPANT_FILE_REPO_KEY) {
```

Comment:
I find the complexity hiding the meaning here. What would you think if all case is handle separetly and then the condition (if or switch) call the 'correct' builder?